### PR TITLE
First version of jenkins nodes enforcing + fixes.

### PR DESCRIPTION
**Pull request by jakubjosef**
First version of jenkins nodes enforcing.
Fixed python-bcrypt dependency definition.
Fixed plurals in state file names.

### CODE REVIEW

**Comment by cloudlab-autobuild** [comment]
Can one of the admins verify this patch?

File: jenkins/client/node.sls [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{% from "jenkins/map.jinja" import client with context %}
+{% for name, node in client.get("node",{}).iteritems() %}
+node_{{ name }}:
```
Comment (fpytloun) [inline jenkins/client/node.sls:3]: I don't see labels enforcing here

File: README.rst [hunk 2]
```diff
@@ -306,6 +306,68 @@ GIT controlled groovy script with shared libraries
...
+.. code-block:: yaml
+
+    jenkins:
+        client:
+          node:
+            node01:
+              name:
+              remote_home: /remote/home/path
+              desc: node-description
+              launcher:
```
Comment (fpytloun) [inline README.rst:355]: Add example on how to add labels

File: README.rst [hunk 3]
```diff
@@ -306,6 +306,68 @@ GIT controlled groovy script with shared libraries
...
+
+Node enforcing from client using JNLP launcher
+
+.. code-block:: yaml
+
+    jenkins:
+        client:
+          node:
+            node01:
+              name:
```
Comment (fpytloun) [inline README.rst]: There should be no name key

**Comment by jakubjosef** [comment]
Changes pushed.

**Pull request merged**